crates/forge/src/scarb.rs

```diff
@@ -36,23 +41,70 @@ pub fn build_contracts_with_scarb(filter: PackagesFilter) -> Result<()> {
 
 pub fn build_test_artifacts_with_scarb(filter: PackagesFilter) -> Result<()> {
     ScarbCommand::new_with_stdio()
-        .arg("snforge-test-collector")
+        .arg("build")
+        .arg("--test")
         .packages_filter(filter)
         .run()
         .context("Failed to build test artifacts with Scarb")?;
     Ok(())
 }
 
+/// collecting by name allow us to dedup targets
+/// we do it because they use same sierra and we display them without distinction anyway
+fn test_targets_by_name(package: &PackageMetadata) -> HashMap<String, &TargetMetadata> {
+    fn test_target_name(target: &TargetMetadata) -> String {
+        // this is logic copied from scarb
+        target
+            .params
+            .get("group-id") // by unit tests grouping
+            .and_then(|v| v.as_str())
+            .map(ToString::to_string)
+            .unwrap_or(target.name.clone()) // else by integration test name
+    }
+
+    package
+        .targets
+        .iter()
+        .filter(|target| target.kind == "test")
+        .map(|target| (test_target_name(target), target))
+        .collect()
+}
+
 pub fn load_test_artifacts(
-    snforge_target_dir_path: &Utf8Path,
-    package_name: &str,
+    target_dir: &Utf8Path,
+    package: &PackageMetadata,
 ) -> Result<Vec<TestTargetRaw>> {
-    let test_artifacts_path =
-        snforge_target_dir_path.join(format!("{package_name}.snforge_sierra.json"));
+    let mut targets = vec![];
+
+    let dedup_targets = test_targets_by_name(package);
+
+    for (target_name, target) in dedup_targets {
+        let tests_location =
+            if target.params.get("test-type").and_then(|v| v.as_str()) == Some("unit") {
+                TestTargetLocation::Lib
+            } else {
+                TestTargetLocation::Tests
+            };
+
+        let target_file = format!("{target_name}.test.sierra.json");
+
+        match read_to_string(target_dir.join(target_file)) {
+            Ok(value) => {
+                let sierra_program = serde_json::from_str::<ProgramArtifact>(&value)?;
```

Comment:
It is a `VersionedProgram` in fact, it works by accident since `VersionedProgram` serializes to 
```
{
  "version": 1,
  // fields of `ProgramArtifact` bcs it is flattened during serialization
} 
```

Reply:
Done in https://github.com/foundry-rs/starknet-foundry/pull/2264